### PR TITLE
Add basic event actions to analytics

### LINKED ISSUE #1701: Add anonymous analytics events to monitor user actions
Reopening the #851 as per this comment: https://github.com/excalidraw/excalidraw/issues/851#issuecomment-625765227

---

We already using [`gtag`](https://developers.google.com/analytics/devguides/collection/gtagjs) so it makes sence to use their events.

As per [`gtag` Events Documentation](https://developers.google.com/analytics/devguides/collection/gtagjs/events) we have the following:

- `<action>` is the string that will appear as the event action in Google Analytics Event reports.
- `<category>` is the string that will appear as the event category.
- `<label>` is the string that will appear as the event label.
- `<value>` is a non-negative integer that will appear as the event value.

Google has some [Default Google Analytics Events](https://developers.google.com/analytics/devguides/collection/gtagjs/events#default_google_analytics_events), we might want to use those or have our own.

Here is the (almost) complete list of things that we can monitor and take actions upon measuring the usage. For example having some thing more prominent or hiding some that are not used that often if at all.

One idea for the label is to have `click` and `shortcut` as separate events to measure how often users are using the shortcuts or not.

More ideas are welcome (feel free to edit this table, by using the [Prettier Extension](https://chrome.google.com/webstore/detail/prettier/fbcfnmplppajblbmdehballiekfgdkhp))

| Name                | Action    | Category | Label | Value |
| ------------------- | --------- | -------- | ----- | :---: |
| Load                |           |          |       |   1   |
| Save                |           |          |       |   1   |
| Export              |           |          |       |   1   |
| Export to PNG       |           |          |       |   1   |
| Export to SVG       |           |          |       |   1   |
| Export to Clipboard |           |          |       |   1   |
| Export to Link      |           |          |       |   1   |
| Export x1 (default) |           |          |       |   1   |
| Export x2           |           |          |       |   1   |
| Export x3           |           |          |       |   1   |
| Reset canvas        |           |          |       |   1   |
| Start collaboration |           |          |       |   1   |
| Selection           | Selection | Tools    |       |   1   |
| Rectangle           | Rectangle | Tools    |       |   1   |
| Diamond             | Diamond   | Tools    |       |   1   |
| Ellipse             | Ellipse   | Tools    |       |   1   |
| Arrow               | Arrow     | Tools    |       |   1   |
| Line                | Line      | Tools    |       |   1   |
| Free draw           | Free draw | Tools    |       |   1   |
| Text                | Text      | Tools    |       |   1   |
| Lock                | Lock      | Tools    |       |   1   |
| Send backward       |           |          |       |   1   |
| Bring forward       |           |          |       |   1   |
| Send to back        |           |          |       |   1   |
| Bring to front      |           |          |       |   1   |
| Duplicate           |           |          |       |   1   |
| Group               |           |          |       |   1   |
| Ungroup             |           |          |       |   1   |
| Copy                |           |          |       |   1   |
| Paste               |           |          |       |   1   |
| Select all          |           |          |       |   1   |
| Delete              |           |          |       |   1   |
| Copy styles         |           |          |       |   1   |
| Paste styles        |           |          |       |   1   |
| Copy as PNG         |           |          |       |   1   |
| Copy as SVG         |           |          |       |   1   |
| Zoom in             |           |          |       |   1   |
| Zoom out            |           |          |       |   1   |
| Zoom reset          |           |          |       |   1   |
| Change language     |           |          |       |   1   |
| Keyboard shortcuts  |           |          |       |   1   |
| Link to GitHub      |           |          |       |   1   |
| Link to E2EE        |           |          |       |   1   |

### Also somehow we need to be able to monitor

- Average number of participants in collaboration mode
- Average size of the local storage

### PR DESCRIPTION
**Pull request by lipis**
### Version 1.0 (this PR)
- [x] MVP [`analytics.md`](https://github.com/excalidraw/excalidraw/blob/events/analytics.md)
- [x] Don't run analytics during tests
- [x] Disable analytics for `localhost`
- [x] Fix tests

### Version 2.0
- Library
- Collaboration
- Export dialog
- Layer
- Align
- Content menu

----

Fixes #1701

### CODE REVIEW

File: src/analytics.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export const EVENT_ACTION = "action";
+
+export const trackEvent = (
+  name: string,
+  category: string,
+  label?: string,
+  value?: number,
+) => {
+  if (gtag) {
+    gtag("event", name, {
```
Comment (lipis) [inline src/analytics.ts:12]: how to properly load (make ts aware of gtag if available) from https://github.com/excalidraw/excalidraw/blob/master/public/index.html#L99

File: src/analytics.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+declare let gtag: Function;
+
+export const EVENT_ACTION = "action";
+export const EVENT_EXIT = "exit";
+export const EVENT_CHANGE = "change";
+export const EVENT_SHAPE = "shape";
+
+export const trackEvent = gtag
```
Comment (lipis) [inline src/analytics.ts:8]: Any idea how to properly use the `gtag` here, that is defined in: https://github.com/excalidraw/excalidraw/blob/events/public/index.html#L87-L99

**Comment by dwelle** [comment]
If needed to debug locally, add `REACT_APP_INCLUDE_GTAG=true` to your `.env.local` file (not in VSC).

File: public/index.html [hunk 3]
```diff
@@ -87,7 +87,7 @@
     <% if (process.env.REACT_APP_INCLUDE_GTAG === 'true') { %>
     <script
       async
-      src="https://www.googletagmanager.com/gtag/js?id=G-H3S0KQSBGX"
+      src="https://www.googletagmanager.com/gtag/js?id=UA-387204-13"
```
Comment (vjeux) [inline public/index.html:90]: What's going on here? Which one is actually being used?

Comment (lipis) [reply public/index.html:90]: It was an experimental checking the latest [Analytics 4](https://analytics.google.com/analytics/web/?hl=en#/p252327942/reports/defaulthome?params=_u..nav%3Ddefault) data.. it's a bit confusing and I think the old look is better.

Comment (lipis) [reply public/index.html:90]: Adding those events to the same analytics account that we have already.

**Pull request merged**